### PR TITLE
Production webpack

### PR DESCRIPTION
**Pull request by coffee-cup**
Addresses https://github.com/prodo-ai/prodo/issues/116

- Add `yarn build` command
- Mock style and file types with Jest
- Autoprefixer with postcss-loader
- split vendor chunks in production
- use `rimraf` instead of `rm -rf` because it works on windows

Many of the webpack changes came from ejected create create app config.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,20 +35,44 @@
...
+    ]
+  },
   "jest": {
     "transform": {
-      "^.+\\.tsx?$": "babel-jest"
+      "^.+\\.(js|jsx|ts|tsx)$": "babel-jest"
+    },
+    "moduleNameMapper": {
+      "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/__mocks__/fileMock.js",
+      "\\.(css|scss|less)$": "identity-obj-proxy"
```
Comment (Onurbon) [inline package.json]: small inconsistency: you're supporting `less` here but the webpack style config is only scanning for ` /\.s?css$/`

File: package.json [hunk 2]
```diff
@@ -34,20 +35,44 @@
...
+      "last 1 safari version"
+    ]
+  },
   "jest": {
     "transform": {
-      "^.+\\.tsx?$": "babel-jest"
+      "^.+\\.(js|jsx|ts|tsx)$": "babel-jest"
+    },
+    "moduleNameMapper": {
+      "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/__mocks__/fileMock.js",
```
Comment (Onurbon) [inline package.json]: any chance we could replace `"<rootDir>/__mocks__/fileMock.js"` by some kind of module, like how you used `identity-obj-proxy` in the next line?

File: tests/__mocks__/fileMock.js [hunk 3]
```diff
@@ -0,0 +1 @@
+module.exports = 'test-file-stub';
```
Comment (Onurbon) [inline tests/__mocks__/fileMock.js]: would be nice to get rid of that if all possible

File: webpack.config.js [hunk 4]
```diff
@@ -1,23 +1,70 @@
...
+      loader: MiniCssExtractPlugin.loader,
+    },
+    "css-loader",
+    "sass-loader",
+    {
+      loader: require.resolve("postcss-loader"),
+      options: {
+        plugins: () => [
+          require("postcss-preset-env")({
+            autoprefixer: {
```
Comment (Onurbon) [inline webpack.config.js]: should we autoprefix in both modes or just when isProduction?

File: webpack.config.js [hunk 5]
```diff
@@ -26,10 +73,7 @@ module.exports = {
         use: ["babel-loader", "source-map-loader"],
         exclude: /node_modules/,
       },
-      {
-        test: /\.s?css$/,
-        loader: ["style-loader", "css-loader", "sass-loader"],
-      },
+      styleLoaders,
       {
         test: /\.(png|jpe?g|svg|ico)$/,
```
Comment (Onurbon) [inline webpack.config.js]: maybe a few things missing for the list used in the jest config  `(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)`?

File: package.json (hunk 2 above)
Comment (coffee-cup) [reply package.json]: I'm actually not sure that `identity-obj-proxy` is correct here for styles. Jest recommends using it for mocking css modules https://jestjs.io/docs/en/webpack#mocking-css-modules. CRA has a mock file for both styles and files.

File: tests/__mocks__/fileMock.js (hunk 3 above)
Comment (coffee-cup) [reply tests/__mocks__/fileMock.js]: We could remove it, but then the user could not test files that import anything but js/ts :/

File: package.json (hunk 1 above)
Comment (coffee-cup) [reply package.json]: Ah good catch. We can probably drop the `less` support.

File: tests/__mocks__/fileMock.js (hunk 3 above)
Comment (coffee-cup) [reply tests/__mocks__/fileMock.js]: Actually, after doing some googling, others have also used the identity-obj-proxy for both css and files

File: webpack.config.js (hunk 4 above)
Comment (coffee-cup) [reply webpack.config.js]: Probably just production

**Pull request merged**